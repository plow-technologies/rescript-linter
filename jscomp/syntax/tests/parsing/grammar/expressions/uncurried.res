let f = (. a, b) => a + b
let f = (. a, . b) => a + b
let f = (. a, b, . c, d) => a + b + c + d

let f = @attr (. a) => @attr2 b => @attr3 (. c) => @attr4 d => ()
let f = (. @attr a, @attr2 b, . @attr3 c, @attr4 d) => ()
let f = (. (@attr a), (@attr2 b), . (@attr3 c), (@attr4 d)) => ()

add(. 1, 2)
add(. 2, 3, 4, . 5, 6, 7, . 8, 9, 10);
