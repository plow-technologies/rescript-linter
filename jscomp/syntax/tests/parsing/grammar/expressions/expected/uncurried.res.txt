let f = ((Function$ (fun a -> fun b -> a + b))[@res.arity 2])
let f = ((Function$ (fun a -> ((Function$ (fun b -> a + b))[@res.arity 1])))
  [@res.arity 1])
let f =
  ((Function$
      (fun a ->
         fun b -> ((Function$ (fun c -> fun d -> ((a + b) + c) + d))
           [@res.arity 2])))
  [@res.arity 2])
let f =
  ((Function$
      ((fun a ->
          ((fun b ->
              ((Function$ ((fun c -> ((fun d -> ())[@res.braces ][@attr4 ]))
                  [@attr3 ]))
              [@res.arity 1]))
          [@res.braces ][@attr2 ]))
      [@attr ]))
  [@res.arity 1])
let f =
  ((Function$
      (fun ((a)[@attr ]) ->
         fun ((b)[@attr2 ]) ->
           ((Function$ (fun ((c)[@attr3 ]) -> fun ((d)[@attr4 ]) -> ()))
           [@res.arity 2])))
  [@res.arity 2])
let f =
  ((Function$
      (fun ((a)[@attr ]) ->
         fun ((b)[@attr2 ]) ->
           ((Function$ (fun ((c)[@attr3 ]) -> fun ((d)[@attr4 ]) -> ()))
           [@res.arity 2])))
  [@res.arity 2])
;;((add 1 2)[@res.uapp ])
;;((((((add 2 3 4)[@res.uapp ]) 5 6 7)[@res.uapp ]) 8 9 10)[@res.uapp ])